### ISSUE TITLE
Non-public markers aren't handled correctly for properties

### ISSUE THREAD

**Issue by zsmb13** [Kotlin/binary-compatibility-validator#36]
When declaring an opt-in annotation and adding that annotation to the `nonPublicMarkers` configuration, properties that have the annotation present are not correctly excluded from the api dump.

Example annotation:

```kotlin
@RequiresOptIn(level = RequiresOptIn.Level.WARNING)
annotation class MyInternalApi
```

A function like this will not show up in the dump (correctly):

```kotiln
@MyInternalApi
fun internalFun() {}
```

However, this property will:

```kotiln
@MyInternalApi
var internalVar = 0
```

... in the form of a getter and a setter:

```plaintext
public static final fun getInternalVar ()I
public static final fun setInternalVar (I)V
```

A workaround that removes these from the API dump is annotating the getter and setter directly, like so:

```kotiln
var otherInternalVar = 0
    @MyInternalApi get
    @MyInternalApi set
```

However, annotating the getter like that doesn't trigger warnings at the call site of the property, so you really need three annotations on each property to satisfy both the plugin and inspections:

```kotlin
@MyInternalApi
var combinedInternalVar = 0
    @MyInternalApi get
    @MyInternalApi set
```

Project with examples and reproducing the issue can be found [here](https://github.com/zsmb13/binary-compat-optin-annotation-issue/).

**Comment by qwwdfsad** [comment]
Thanks for the report, this is indeed a bug in binary compatibility validator.
Semantically, `@MyInternalApi` should be applied to both getter and setter (and compiler behaves as it does), but in the implementation, we do not redundantly mark each getter and setter and thus BCV treats them like part of API.

PRs are welcome, for the reference we got pretty similar issue with `@PublishedApi` and default parameters: https://github.com/Kotlin/binary-compatibility-validator/pull/30

**Comment by martinbonnin** [comment]
Another issue is that if `@MyInternalApi` is an "Opt-in" marker, it's not possible to add it on the getter. The kotlin compiler fails with:

```
Opt-in requirement marker annotation cannot be used on getter
```

That makes it harder to workaround the issue because "Opt-in" markers are typically the kind of annotation that are nice to hide.

**Comment by qwwdfsad** [comment]
Unfortunately, we're overwhelmed with other work right now, so we can't provide ETA for the fix. 
PRs will be timely reviewed though

**Comment by martinbonnin** [comment]
Tentative PR there: https://github.com/Kotlin/binary-compatibility-validator/pull/81. It pulls all the property/field annotations on getters/setters, which I think is what we want? 
But I have very limited experience about the semantics of fields vs setters vs property annotations, so I might be missing something. All feedback welcome!
